Reserveringssysteem/CancelReservation.cs

```diff
@@ -6,68 +6,88 @@
-                    input = Console.ReadLine();
-                    if (input == "")
-                        break;
-                }
                 if (input.ToLower().Length == 4)
                 {
-                    ReservationsList = Deserialize<List<Reservation>>("reservations.json");          
-                    for (int i = 0; i < ReservationsList.Count; i++)                            //Loopt door de reservation items om te kijken of de reserverings code erin staat
+                    ReservationsList = Deserialize<List<Reservation>>("reservations.json");
+                    for (i = 0; i < ReservationsList.Count; i++)                            //Loopt door de reservation items om te kijken of de reserverings code erin staat
                     {
-                        if (ReservationsList[i].ReservationId.ToLower() == input.ToLower())     //Als een match gevonden wordt, item verwijderen
+                        if (ReservationsList[i].ReservationId.ToLower() == input.ToLower())     //Als een match gevonden wordt...
                         {
                             Console.Clear();
                             CancelTitle();
-                            Console.WriteLine("De code staat in het systeem en de reservering wordt nu geannuleerd..");
-                            foundItem = true ;
-                            ReservationsList.Remove(ReservationsList[i]);                      // Verwijderd bijbehorende item uit JSON
-                            Serialize(ReservationsList,"reservations.json");                   // Slaat de JSON opnieuw op na de aanpassing
+                            AreUSure();                                                         // Laat het keuzemenu zien; ja voor verwijderen <--> nee voor terug naar vorige scherm
+                            keepTrue = false;
                             break;
                         }
                     }
-                    }
-                if (foundItem == false)                                                // Als er geen match is vragen of je het opnieuw wilt proberen
-                {
-                    Console.Clear();
-                    CancelTitle();
-                    Console.WriteLine("\nReserveringscode niet herkend.. Probeer opnieuw of ga terug naar het hoofdmenu door op 'enter' te drukken: ");
-                    input = Console.ReadLine();
-                    if (input == "")
-                        break;
                 }
-                else if (foundItem)                                                     // Sluit af als reservering geannuleerd is
+                else if (input == "")
                 {
-                    Console.WriteLine("\nDe reservering is geannuleerd!");
+                    keepTrue = false;
                     Utils.Enter();
-                    break;
+                }
+                else if (input.Length != 4 && input != "")
```

Comment:
Volgens mij kan je van deze else if gewoon een else maken. Want als de code hier komt is de input sowieso geen 4 karakters (dan zou hij stoppen bij de eerste if statement) en het is ook geen lege string (anders zou hij stoppen bij de eerste else if). Maar wat je nu hebt is zeker niet verkeerd. Ik denk alleen dat een else wat beter is.